test/unit/rx-query.test.ts

```diff
@@ -1582,4 +1584,254 @@ describe('rx-query.test.ts', () => {
+        });
+
+        it('removing an item from the database, but not from cache does not lead to wrong results after restoring', async () => {
+            const {collection} = await setUpPersistentQueryCacheCollection();
+
+            const human1 = schemaObjects.human('1', 30);
+            const human2 = schemaObjects.human('2', 40);
+            const human3 = schemaObjects.human('3', 50);
+
+            await collection.bulkInsert([human1, human2, human3]);
+
+            const query1 = collection.find({ limit: 2, sort: [{age: 'asc'}] });
+            const queryId = query1.persistentQueryId();
+            const lwt = now();
+
+            const cache = new Cache();
+            await cache.setItem(`qc:${queryId}`, [human1.passportId, human2.passportId, human3.passportId]);
+            await cache.setItem(`qc:${queryId}:lwt`, `${lwt}`);
+
+            const removeQuery = collection.find({ selector: { passportId: '2' }});
+            await removeQuery.remove();
+
+            clearQueryCache(collection);
+
+            const query2 = collection.find({ limit: 2, sort: [{age: 'asc'}] });
+            query2.enablePersistentQueryCache(cache);
+
+            assert.strictEqual(cache.getItem(`qc:${queryId}`).length, 3);
+
+            const result2 = await query2.exec();
+
+            assert.strictEqual(query1._execOverDatabaseCount, 0);
+            assert.strictEqual(query2._execOverDatabaseCount, 0);
+            assert.deepStrictEqual(result2.map(item => item.passportId), ['1', '3']);
+
+            collection.database.destroy();
+        });
+
+        it('old cache values are updated when documents are modified', async () => {
+            const {collection} = await setUpPersistentQueryCacheCollection();
+
+            const human1 = schemaObjects.human('1', 30);
+
+            await collection.bulkInsert([human1]);
+
+            // fill cache
+            const cache = new Cache();
+            const query1 = collection.find({});
+            query1.enablePersistentQueryCache(cache);
+            const queryId = query1.persistentQueryId();
+
+            const result1 = await query1.exec();
+            assert.strictEqual(result1.length, 1);
+            assert.strictEqual(cache.size, 2);
+
+            clearQueryCache(collection);
+
+            // go back in time
+            const lwt = now() - 7200 * 1000; // go back in time (2hrs)
+            await cache.setItem(`qc:${queryId}:lwt`, `${lwt}`);
+
+            const query2 = collection.find({});
+            query2.enablePersistentQueryCache(cache);
+            await query2._persistentQueryCacheLoaded;
+
+            await result1[0].modify(data => {
+              data.age = 40;
+              return data;
+            });
+
+            await query2.exec();
+
+            const currLwt = Number(await cache.getItem(`qc:${queryId}:lwt`));
+            assert.strictEqual(currLwt > lwt, true);
+
+            collection.database.destroy();
```

Comment:
can't you put these in a beforeAfter()?

Reply:
maybe kind of annoying to do with a global variable. idc that much

Reply:
Would be nice. I don't even know which test library RxDB uses. I just copy & pasted code from other tests.